woocommerce/woocommerce-gateway-stripe
# Fix E2E tests for the new checkout experience

Fixes #3092, #3120

## Changes proposed in this Pull Request:

The PR https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3106 refactored the tests to have two groups: one for the `legacy checkout experience` and another for the `new checkout experience` (now the default for new installations). However, to keep that PR as concise as possible, it did not add tests for the `new checkout experience`.

This PR builds on that and adds the tests for the `new checkout experience`:

![Screenshot_2024-05-08_at_4_48_09 PM](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/c1b0b35d-0c64-4d2e-b11b-b6b94afeaf6e)

## Testing instructions

Confirm all of the E2E tests  [running](https://github.com/woocommerce/woocommerce-gateway-stripe/actions/runs/9083744075/job/24963085253?pr=3117) in this PR pass:

![Screenshot 2024-05-14 at 5 37 01 PM](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/433d6714-835f-4bf5-beb1-c01b556b4728)

**_[Optional]_** The tests can be executed in docker:
1. Fetch this branch
    ```
    git checkout 3092/add-new-checkout-e2e-tests
    ```
2. Install dependencies:
    ```
    npm install
    ```
3. Build the JS (**Don't skip this step**; it's required because of the renames from `wcpay` -> `wcstripe`)
    ```
    npm run build
    ```
4. Make sure docker is running, and initialize the E2E docker environment
    ```
    npm run test:e2e-setup
    ```
5. Run the test suite
    ```
    npm run test:e2e
    ````
6. All tests should pass
    ![Screenshot 2024-05-14 at 6 36 30 PM](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/0610ca2a-b274-4a26-a549-14d77b115df0)
